### DISCUSSION TITLE
Matter for esp?

### DISCUSSION THREAD

**Discussion by toshibochan** [RavenSystem/esp-homekit-devices#1751] [General]
https://www.espressif.com/en/news/Matter_on_ESP32

Anyone is developing matter for esp for homekit?

**Comment by toshibochan** [comment]
https://github.com/esphome/feature-requests/issues/1430

**Comment by toshibochan** [comment]
https://openthread.io/vendors/espressif

**Comment by RavenSystem** [comment]
Watch this video (you can read good subtitles is you watch it on a desktop computer): https://youtu.be/GO8T2174f7M

And you will understand why HAA will never be Matter compatible.

**Reply by makdiuk** [reply]
So funny and very clear view to revolutionary matter - thanks 😄

**Reply by borissov** [reply]
The biggest difference is that Matter is open standard..

**Reply by RavenSystem** [reply]
HomeKit is an open standard too. And thanks to this, HAA is possible.

**Comment by tokamac** [comment]
**Tasmota** (which was said to never be compatible with HomeKit) is [adding Matter compatibility to ESP32](https://tasmota.github.io/docs/Matter/) from version 12.5.0 Peyton, making it… finally compatible with HomeKit. The first implementation is compatible with lights (1/2/3 channels) and switches.

**Reply by RavenSystem** [reply]
I think you are confusing HomeKit with Apple Home. Tasmota is not HomeKit, and Matter is not compatible with HomeKit. Matter and HomeKit are very different standards, but both compatible with Apple Home. Tasmota is Matter compatible, and HAA is HomeKit.

Tasmota (which was said to never be compatible with HomeKit) keeps incompatible with HomeKit, in the same way HAA keeps incompatible with Matter. But both (Matter and HomeKit) are supported by Apple Home. https://developer.apple.com/apple-home/

**Reply by tokamac** [reply]
You're right, this was a confusion I made, sorry. HomeKit ≠ Matter, yes. I was indeed speaking, like an end user, of the Apple Home app (and Siri) that is compatible with HomeKit accessories, and now also with Matter accessories. It is my understanding that after a period of transition, when Matter will be mature enough (it is not yet, not as much as HomeKit at the moment) then HomeKit will become deprecated, to finally even cease to exist, letting only Matter products remaining on the market.

The most important and new thing in this announcement is that accessories flashed with Tasmota will finally be able to be used by Apple users (in the Home app, via Matter) and doing so, these Tasmota Matter devices will also work seamlessly with existing HomeKit devices (in the Home app and Siri, via an edge router such as an Apple TV or HomePod).

This is a pretty big announcement to be honest, because the list of devices that can be flashed with Tasmota is quite huge (even if only ESP32 will be compatible. Nevertheless ESP8266 devices controlled by an ESP32 Matter bridge may be coming also). This will also give the opportunity for DIYers to design a single accessory and target all ecosystems at once with it, with that device able to be used by anyone, including in the Apple ecosystem, but not only restricted to the Apple ecosystem.

**Reply by RavenSystem** [reply]
HomeKit will not be deprecated because Apple will not open Secure Video, Apple keys, Adaptive lightning... and all HomeKit features. And HomeKit is better standard than Matter.

**Reply by tokamac** [reply]
Very true for Apple Keys indeed, a very Apple-centric feature that is owned and developed exclusively by Apple. 
As for HSV, true in its current form obviously, but I cannot see a definitive reason why Apple could not implement some "MSV" (once cameras are part of the Matter standard, which they are not yet) since Secure Video is a software service bound to their iCloud subscriptions and not really to the device itself.
Also, I am confident that Adaptive Lightning (or whatever it will be called, "circadian rhythm" or anything else) which is not a technology that Apple has a monopoly on (it already exists elsewhere) will comme to Matter lightings at some point too. 

Yes, HomeKit is superior to Matter… for now.

Just put yourself in the shoes of a manufacturer of home automation devices, if at some point in the short future Matter (which is an evolving standard, today perfectible and incomplete, but only in its early childhood) becomes on par with HomeKit from the point of view of device types and features: then, why target only Apple customers when you can target them PLUS customers from the other side who are using Amazon, Google or Samsung home ecosystems… all at the same time? When it's the case, nobody will make HomeKit-only devices on the market anymore. Eve has understood it well.

**Reply by xrust83** [reply]
Hello, allow me to express my opinion. I think Matter won't make it to the masses due to a number of factors, these devices, at this stage, are more expensive, the idea is fresh at first glance so that people with iPhone and Android can control the same devices, but it works in public places, in the family if dad has - apple then after a while everyone has an apple :-), double ecosystems are not needed. 2. If you are building a smart home on Apple Home, why would you pay more for a Matter that you don't need. 3. And the main reason for the dead end of this development of “technology” is to allow android to turn on the light bulb in apple home is one thing, but to put android in homekit is another, we will get primitive devices with limited functions in an alternative ecosystem, a war will begin for the user, on an iPhone you can turn on light, and if you buy an android, the light will flash to the music, or vice versa. And now we are talking about the imaginary compatibility of two ecosystems that have been running for more than 15 years, specially, different adapters for charging 😂, there is no doubt that they will become friends 😜

**Reply by tokamac** [reply]
1. Matter products are not more expensive at all. Actually it's quite the contrary. Why? Because manufacturers pay a premium price to Apple for everything MFi-related (the infamous "Apple tax"). In fact they even overdo it more on the final price point just with respect to Apple customers because they know that this specific target is willing to pay more than "stingy" PC & Android users (not a belief but a fact demonstrated by many independent studies about hardware and paid apps). But enough theory. The best concrete demonstration is that the cheapest smart lightning currently available on the market among HomeKit-only, Alexa/Google-only, made by Aqara/Mi, Philips Hue, etc. happens to be the single one available that is… Matter compatible: WiZ GU10 DIMMABLE spot, sold at 8.33 € VAT excl. in the EU.

2. It is important, considering a market, a technology and how they are about to evolve together, to think globally from various points of view and not to remain focused on its personal experience only. In this regard:
- Switch happens (in both directions). Your "solution" just locks customers a bit more firmly into a single ecosystem and adds more barriers to any possible future personal switch.
- Not everyone is a tenant in their home. Owners sell their property to make capital gains. A home automation system adds value to the property. No one wants to target only a fraction of potential buyers.
- You are concluding about your own personal situation, as a particular individual, and as a customer. Yet customerS are all different and have various needs. But this is not event the point. These products are not made by customers, they are made by manufacturers. And as I explained already before your post, when designing a technological product, for similar development and sales costs between two standards to choose from, a manufacturer has every interest in choosing to target the maximum number of potential customers, rather than a fraction of the population. Simple math. Another concrete example is Eve's strategy, as explained also as an example above.

About Matter, one should skate to where the puck is going to be, not to where it is currently. Meaning that all my talk is in the prospects, not the current apparent status quo at time T.

**Reply by RavenSystem** [reply]
I am HomeKit and Matter certified devices developer. I am into "manufacturers shoes", and I call tell you that Matter is only good for manufacturers into CSA. Third-party manufacturers must pay $7000 only to start development, and then, a lot of more for Matter certifications. In other way, the infamous "Apple tax" is a fake. The initial Apple price is $99 (Matter is $7000), and I can not tell you how much money cost to certificate an Apple HomeKit device, but I will tell you that I could pay it with the money I have in my hand just now. Manufacturers are adopting Matter because market says that they must, but then they must keep HomeKit because Matter lacks tons of features.

Other thing, Apple will not give their technologies to Amazon and Google to allow Matter to use them. Secure Video, Apple Keys and other Apple features will keep in Apple.

The reality is that manufacturers could add Alexa and Google Home support for free, paying only for HomeKit, and now, they must pay a lot for Matter, having many problems because Matter technology needs much hardware power.

**Reply by tokamac** [reply]
Thank you for this explanation. Fair enough, I agree with your point of view considering these facts and will end here my digression :)

The Matter fee is amazing. Jaw-dropping. In this regard, I am wondering how Wiz can even sell Matter-compatible spots at such a low price point. Surely because they are a sister company from giant Signify and can absorb such initial costs.

A last precision for HomeKit Secure Video, because here my point was not at all to "give it to Amazon and Google", I may have badly explained. I just think that Apple could keep Secure Video for its own ecosystem, not sharing it, but allow at some point Matter cameras (when the spec allows it and such devices are readily available) to be used with it, inside the Home app exclusively. They could because HSV is a software subscription service separated from the hardware, locked into iCloud. The same Matter camera bought by an Android user instead could not benefit from such hypothetical "Apple Matter Secure Video" on a Google Home device. That's all. 

Thanks again for the insight, much appreciated. It changed my mind somehow.
